app/src/main/java/com/example/myapplication/vm/LoginViewModel.kt

```diff
@@ -0,0 +1,46 @@
+package com.example.myapplication.vm
+
+import android.app.Application
+import android.content.Context
+import android.content.SharedPreferences
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.MutableLiveData
+
+class LoginViewModel (app: Application): AndroidViewModel(app) {
+
+    val email = MutableLiveData<String?>()
```

Comment:
Also, is better not to expose a mutablelivedata to the view, because it gives a way to trigger unexpected events. A good practice is to expose a LiveData that references to a MutableLiveData. 
Example:
 private val _loginError: MutableLiveData<Boolean> = MutableLiveData(...)
 val loginError: LiveData<...>
        get() = _loginError